LayoutTests/http/tests/cache/cache-match-invalid-url.html

```diff
@@ -0,0 +1,19 @@
+<script>
+if (window.testRunner) {
+    testRunner.dumpAsText();
+    testRunner.waitUntilDone();
+}
+
+function testFunc() {
+  caches.open('a').then((cache) => {
+    cache.match(`/\\X\u0600`);
```

Comment:
We could try and catch here, and verify the exception is expected (e.g. url is invalid).
Maybe something like:
try {
        cache.match(`/\\X\u0600`);
        assert_unreached("Expected exception to be thrown");
} catch (err) {
        assert_equals(err.code, DOMException.INVALID_STATE_ERR, "The exception message");
}
Will need to import some helper js files as other tests in cache-storage/.

Reply:
This function returns a promise, therefore, it will reject the promise, not throw an exception, unless we make the function async and use `await`.

Reply:
Good point, I'll update the test so that it can validate the exception

Reply:
Based on https://developer.mozilla.org/en-US/docs/Web/API/DOMException#error_names, the exception object is of the type TypeError